plone/app/vocabularies/search.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+from Products.CMFPlone import PloneMessageFactory as _
+from zope.interface import implementer
+from zope.schema.interfaces import IVocabularyFactory
+from zope.schema.vocabulary import SimpleTerm
+from zope.schema.vocabulary import SimpleVocabulary
+
+
+@implementer(IVocabularyFactory)
+class SearchIndexesVocabulary(object):
+    """Vocabulary factory for search indexes.
+
+      >>> from zope.component import queryUtility
+      >>> from plone.app.vocabularies.tests.base import create_context
+
+      >>> name = 'plone.app.vocabularies.SearchIndexes'
+      >>> util = queryUtility(IVocabularyFactory, name)
+      >>> context = create_context()
+
+      >>> indexes = util(context)
+      >>> indexes
+      <zope.schema.vocabulary.SimpleVocabulary object at ...>
+
+      >>> len(indexes.by_token)
+      3
+
+      >>> term = indexes.by_token['Date']
+      >>> term.title, term.token, term.value
+      (u'date (newest first)', 'Date', 'Date')
+    """
+
+    def __call__(self, context):
+        items = []
+        items.append((_(u'relevance'), 'relevance'))
+        items.append((_(u'date (newest first)'), 'Date'))
+        items.append((_(u'alphabetically'), 'sortable_title'))
```

Comment:
I'am ok adding a sortindexes vocabulary. This can be used in folder 
contents, collections...

This is a general feature.

It should be extensible, so registry is probably the place you  want to 
store these vocabulary.


Am 24. Juni 2016 9:32:10 nachm. schrieb Héctor Velarde 
<notifications@github.com>:

>> +      >>> indexes
>> +      <zope.schema.vocabulary.SimpleVocabulary object at ...>
>> +
>> +      >>> len(indexes.by_token)
>> +      3
>> +
>> +      >>> term = indexes.by_token['Date']
>> +      >>> term.title, term.token, term.value
>> +      (u'date (newest first)', 'Date', 'Date')
>> +    """
>> +
>> +    def __call__(self, context):
>> +        items = []
>> +        items.append((_(u'relevance'), 'relevance'))
>> +        items.append((_(u'date (newest first)'), 'Date'))
>> +        items.append((_(u'alphabetically'), 'sortable_title'))
>
> @rodfersou move this to 
> https://github.com/plone/plone.app.controlpanel/pull/63/files it would be 
> easier to finish and mantain; as mentioned before: we don't need this named 
> vocabulary in this package; we just need a static vocabulary.
>
> and BTW, this should not be named `SearchIndexesVocabulary`; there's 
> nothing about search indexes in here; this is about how to sort search results.
>
> ---
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/plone/plone.app.vocabularies/pull/32/files/8f325adb033727f1debc61de686d3235e89e024a#r68450625